### PR TITLE
bugfixes for findfirst in MixedDofHandler+ tests

### PR DESCRIPTION
**Pull request by kimauth**
`findfirst` returns `nothing` when it doesn't find a match, not 0

### CODE REVIEW

File: src/Dofs/MixedDofHandler.jl [hunk 1]
```diff
@@ -98,7 +98,7 @@ function getfielddim(dh::MixedDofHandler, name::Symbol)
 
     for fh in dh.fieldhandlers
         field_pos = findfirst(i->i == name, getfieldnames(fh))
-        if field_pos > 0
+        if field_pos != nothing
```
Comment (fredrikekre) [inline src/Dofs/MixedDofHandler.jl:101]:
```suggestion
        if field_pos !== nothing
```

File: src/Dofs/MixedDofHandler.jl [hunk 2]
```diff
@@ -388,7 +388,7 @@ end
 
 function find_field(fh::FieldHandler, field_name::Symbol)
     j = findfirst(i->i == field_name, getfieldnames(fh))
-    j == 0 && error("did not find field $field_name")
+    j == nothing && error("did not find field $field_name")
```
Comment (fredrikekre) [inline src/Dofs/MixedDofHandler.jl:391]:
```suggestion
    j === nothing && error("did not find field $field_name")
```

File: src/Export/VTK.jl [hunk 3]
```diff
@@ -110,7 +110,7 @@ function WriteVTK.vtk_point_data(vtkfile, dh::MixedDofHandler, u::Vector, suffix
         for fh in dh.fieldhandlers
             # check if this fh contains this field, otherwise continue to the next
             field_pos = findfirst(i->i == name, getfieldnames(fh))
-            if field_pos == 0 && continue end
+            if field_pos == nothing && continue end
```
Comment (fredrikekre) [inline src/Export/VTK.jl:113]:
```suggestion
            if field_pos === nothing && continue end
```

**Review by fredrikekre — COMMENTED** [review]
Compiler likes `===` when comparing with `nothing`.

File: src/Export/VTK.jl [hunk 4]
```diff
@@ -110,7 +110,7 @@ function WriteVTK.vtk_point_data(vtkfile, dh::MixedDofHandler, u::Vector, suffix
         for fh in dh.fieldhandlers
             # check if this fh contains this field, otherwise continue to the next
             field_pos = findfirst(i->i == name, getfieldnames(fh))
-            if field_pos == 0 && continue end
+            if field_pos === nothing && continue end
```
Comment (fredrikekre) [inline src/Export/VTK.jl:113]: Not related to your changes, but I found this difficult to read. Perhaps:
```suggestion
            field_pos === nothing && continue
```

Comment (kimauth) [reply src/Export/VTK.jl:113]: I've also been thinking if it would be better to replace this by some `hasfield(fh, fieldname)`.

**Pull request merged**